cmake/modules/packages/FindCryptoPP.cmake

```diff
@@ -77,8 +77,26 @@ if(CRYPTOPP_FOUND)
     if(NOT TARGET CRYPTOPP::CRYPTOPP)
         add_library(CRYPTOPP::CRYPTOPP UNKNOWN IMPORTED)
         set_target_properties(CRYPTOPP::CRYPTOPP PROPERTIES
-                              INTERFACE_INCLUDE_DIRECTORIES "${CRYPTOPP_INCLUDE_DIR}"
-                              IMPORTED_LINK_INTERFACE_LANGUAGES "CXX"
-                              IMPORTED_LOCATION "${CRYPTOPP_LIBRARY}")
+                INTERFACE_INCLUDE_DIRECTORIES "${CRYPTOPP_INCLUDE_DIR}"
+                IMPORTED_LINK_INTERFACE_LANGUAGES "CXX")
+
+        if(CRYPTOPP_LIBRARY_RELEASE)
+            set_property(TARGET CRYPTOPP::CRYPTOPP APPEND PROPERTY
+                    IMPORTED_CONFIGURATIONS RELEASE)
+            set_target_properties(CRYPTOPP::CRYPTOPP PROPERTIES
+                    IMPORTED_LOCATION_RELEASE "${CRYPTOPP_LIBRARY_RELEASE}")
+        endif()
+
+        if(CRYPTOPP_LIBRARY_DEBUG)
+            set_property(TARGET CRYPTOPP::CRYPTOPP APPEND PROPERTY
+                    IMPORTED_CONFIGURATIONS DEBUG)
+            set_target_properties(CRYPTOPP::CRYPTOPP PROPERTIES
+                    IMPORTED_LOCATION_DEBUG "${CRYPTOPP_LIBRARY_DEBUG}")
+        endif()
+
+        if(NOT CRYPTOPP_LIBRARY_RELEASE AND NOT CRYPTOPP_LIBRARY_DEBUG)
+            set_target_properties(CRYPTOPP::CRYPTOPP PROPERTIES
+                    IMPORTED_LOCATION_RELEASE "${CRYPTOPP_LIBRARY}")
```

Comment:
IMPORTED_LOCATION? (Does this case exist at all?)

Reply:
> (Does this case exist at all?)

Presumably not, looking at https://cmake.org/cmake/help/latest/module/SelectLibraryConfigurations.html. I've removed that case, and the original in FindDeflate.cmake from which I copied&pasted

Reply:
CMake's `FindZLIB.cmake` has similar code, however it doesn't even look for for _DEBUG/_RELEASE if `ZLIB_LIBRARY` is already set. I guess it is mostly backward compatibility, but it could also be single-config convenience. In any case, it is not a visible cache variable unless created by the user.